openshift/README.md

```diff
@@ -0,0 +1,207 @@
+
+Our application template take in the following parameters:
+
+| Name | Required | Description |
+| --- | --- | --- |
+| REPO_NAME | yes | Application repository name |
+| JOB_NAME | yes | Job identifier (i.e. 'pr-5' OR 'master') |
+| NAMESPACE | yes | which namespace/"environment" are we deploying to? dev, test, prod? |
+| APP_NAME | yes | short name for the application |
+| ROUTE_HOST | yes | base domain for the publicly accessible URL |
+| ROUTE_PATH | yes | base path for the publicly accessible URL |
+
+The Jenkins pipeline will handle deployment invocation automatically. However should you need to run it manually, you can do so with the following for example:
+
+```sh
+export NAMESPACE=<yournamespace>
+export APP_NAME=<yourappshortname>
+
+oc process -n $NAMESPACE -f openshift/app.dc.yaml -p REPO_NAME=common-hosted-form-service -p JOB_NAME=master -p NAMESPACE=$NAMESPACE -p APP_NAME=$APP_NAME -p ROUTE_HOST=$APP_NAME-dev.pathfinder.gov.bc.ca -p ROUTE_PATH=master -o yaml | oc apply -n $NAMESPACE -f -
+```
+
+Due to the triggers that are set in the deploymentconfig, the deployment will begin automatically. However, you can deploy manually by use the following command for example:
+
+```sh
+oc rollout -n $NAMESPACE latest dc/<buildname>-master
+```
+
+*Note: Remember to swap out the bracketed values with the appropriate values!*
+
+## Pull Request Cleanup
+
+As of this time, we do not automatically clean up resources generated by a Pull Request once it has been accepted and merged in. This is still a manual process. Our PR deployments are all named in the format "pr-###", where the ### is the number of the specific PR. In order to clear all resources for a specific PR, run the following two commands to delete all relevant resources from the Openshift project (replacing `PRNUMBER` with the appropriate number):
+
+```sh
+export NAMESPACE=<yournamespace>
+export APP_NAME=<yourappshortname>
+
+oc delete all,secret,pvc,nsp,rolebinding -n $NAMESPACE --selector app=$APP_NAME-pr-<PRNUMBER>
+oc delete all,svc,cm,sa,role,secret -n $NAMESPACE --selector cluster-name=pr-<PRNUMBER>
+```
+
+The first command will clear out all related executable resources for the application, while the second command will clear out the remaining Patroni cluster resources associated with that PR.
+
+## Appendix - Supporting Deployments
+
+There will be instances where this application will need supporting modifications or deployment such as databases and business analytics tools. Below is a list of initial reference points for other Openshift templates that could be leveraged and bolted onto the existing Jenkins pipeline if applicable.
+
+### Metabase
+
+- [Overview & Templates](https://github.com/bcgov/nr-get-token/wiki/Metabase)
+
+### MongoDB
```

Comment:
Not using mongo in this one right?

Reply:
This is just copied over from comfort verbatim. This general appendix section is more just a pointer/reference for "if you need other templates, here's a pointer". Either way, just realized the link is busted - so just fixed that.